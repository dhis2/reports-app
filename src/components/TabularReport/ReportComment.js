import React from 'react'
import PropTypes from 'prop-types'
import { InputField, Button } from '@dhis2/d2-ui-core'
import i18n from '@dhis2/d2-i18n'

const isCommentingActionEnabled = comment => comment.trim()
const actionButtonPlaceholder = i18n.t('Share')
const inputFieldPlaceholder = i18n.t(
    'Write a comment, question or interpretation of this report'
)

export const ReportComment = props => (
    <div id="share-component">
        <div>
            <InputField
                placeholder={inputFieldPlaceholder}
                type="text"
                multiline
                fullWidth
                value={props.comment}
                onChange={props.setDataSetReportComment}
            />
            <Button
                raised
                color="primary"
                onClick={props.shareDataSetReportComment}
                disabled={!isCommentingActionEnabled(props.comment)}
            >
                {actionButtonPlaceholder}
            </Button>
        </div>
        <style jsx>{`
            div > div {
                margin: 16px 0;
            }
        `}</style>
    </div>
)

ReportComment.propTypes = {
    comment: PropTypes.string.isRequired,
    setDataSetReportComment: PropTypes.func.isRequired,
    shareDataSetReportComment: PropTypes.func.isRequired,
}
