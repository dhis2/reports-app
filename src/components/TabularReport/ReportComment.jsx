import i18n from '@dhis2/d2-i18n'
import { InputField, Button } from '@dhis2/d2-ui-core'
import PropTypes from 'prop-types'
import styles from './ReportComment.module.css'

const isCommentingActionEnabled = (comment) => comment.trim()
const actionButtonPlaceholder = i18n.t('Share')
const inputFieldPlaceholder = i18n.t(
    'Write a comment, question or interpretation of this report'
)

export const ReportComment = (props) => (
    <div id="share-component">
        <div className={styles.wrapper}>
            <InputField
                placeholder={inputFieldPlaceholder}
                type="text"
                multiline
                fullWidth
                value={props.comment}
                onChange={props.setDataSetReportComment}
            />
            <Button
                raised
                color="primary"
                onClick={props.shareDataSetReportComment}
                disabled={!isCommentingActionEnabled(props.comment)}
            >
                {actionButtonPlaceholder}
            </Button>
        </div>
    </div>
)

ReportComment.propTypes = {
    comment: PropTypes.string.isRequired,
    setDataSetReportComment: PropTypes.func.isRequired,
    shareDataSetReportComment: PropTypes.func.isRequired,
}
