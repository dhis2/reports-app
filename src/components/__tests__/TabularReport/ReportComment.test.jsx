import { InputField, Button } from '@dhis2/d2-ui-core'
import { shallow } from 'enzyme'
import React from 'react'
import { ReportComment } from '../../TabularReport/ReportComment.jsx'

describe('<ReportComment/>', () => {
    const baseProps = {
        comment: 'I want to comment on this',
        shareDataSetReportComment: jest.fn(),
        setDataSetReportComment: jest.fn(),
    }

    it('renders the InputField with the given comment', () => {
        const wrapper = shallow(<ReportComment {...baseProps} />)
        const input = wrapper.find(InputField)

        expect(input.exists()).toBe(true)
        expect(input.prop('value')).toBe(baseProps.comment)
    })

    it('renders a Button with text "Share"', () => {
        const wrapper = shallow(<ReportComment {...baseProps} />)
        const button = wrapper.find(Button)

        expect(button.exists()).toBe(true)
        expect(button.children().text()).toBe('Share')
    })

    it('disables the button if comment is empty', () => {
        const wrapper = shallow(<ReportComment {...baseProps} comment="" />)
        const button = wrapper.find(Button)
        expect(button.prop('disabled')).toBe(true)
    })

    it('enables the button if comment is non-empty', () => {
        const wrapper = shallow(<ReportComment {...baseProps} />)
        const button = wrapper.find(Button)
        expect(button.prop('disabled')).toBe(false)
    })

    it('calls setDataSetReportComment when InputField changes', () => {
        const wrapper = shallow(<ReportComment {...baseProps} />)
        wrapper.find(InputField).simulate('change', { value: 'new comment' })
        expect(baseProps.setDataSetReportComment).toHaveBeenCalledWith({
            value: 'new comment',
        })
    })

    it('calls shareDataSetReportComment when Button is clicked', () => {
        const wrapper = shallow(<ReportComment {...baseProps} />)
        wrapper.find(Button).simulate('click')
        expect(baseProps.shareDataSetReportComment).toHaveBeenCalled()
    })
})
