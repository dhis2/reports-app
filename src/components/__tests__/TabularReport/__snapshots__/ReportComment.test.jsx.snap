// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReportComment/> Matches the snapshot 1`] = `
<div
  id="share-component"
>
  <div>
    <TextField
      fullWidth={true}
      multiline={true}
      onChange={[MockFunction]}
      placeholder="Write a comment, question or interpretation of this report"
      type="text"
      value="I want to comment on this"
    />
    <Button
      color="primary"
      disabled={false}
      onClick={[MockFunction]}
      raised={true}
    >
      Share
    </Button>
  </div>
  <style>
    
            div &gt; div {
                margin: 16px 0;
            }
        
  </style>
</div>
`;
